neural_structured_learning/examples/notebooks/adversarial_cnn_transfer_learning_fashionmnist.ipynb

```diff
@@ -0,0 +1,3081 @@
z1TcXp6bGmMnKjYQVqezq3+Q9E7HRCqzFH/PrsrMC91De6imhttDByom7jhA6+ryvRXr5KPHEdZqxat+Drb98cw4deqzj9iekTqzL5d0dvpevpHmNTPN7+8Vfzuk2N55j+2nOt4z9W+Svp7Oti1TvBxyl/WW1IzzJD3N8b7SabbfoPg7Wb3n7c2pzbOpYn11TuXqqKqLFX9n3+WY9OX4NPwnuYU7JtTYQS0ZAB3vP/2J4sGmL2amHT5wunD0j9lnHazUYDxRsQd5o6TWh9W9X/Hax8sdM8X8WPFMiiTJMePJcIaaBxSPiN6meF3ov0t6Zwjh0hCTArxLcQNcqZiQ4bvjCPV4xdPjdyuu6P+plF2geMnF7xVPbT6q9S9R6ZSjFSvLaxU/1zmK1/BKsTK+QLHS/rVipfgkx0w056e3jyleCvRjxfV4teLRoGNS+WcUs/cMXwc95huiQwjXSDpOsXJamuKsPtvmfYrfx+oUc1ee1ZEuI/w/ip2TpYoV6htbRjtX8RKmKxVv0jxVejLJxYMemeTiEcXT11I8Sv/IcIFjtqfqDvxGxcukVireuPzaEMKyFNcPFbfdnypuy7dqZIc2Oy16aoGk/wkh3BZixq67Q8zQdpLivRBrJb1acZ+6T9Ib1LJPpkbCZYqXbn29Mvx2xbNaH1T8Mbxd0j8oU59PtL4LIVyrmKXuMsVOzp9J+vlYpy8YbV8/QdLpjjfrvz6EcL1iw+CWNGx7SZ9V/Cw/sr1asT56ZgOxTcRJHvm8lLFc9rhS8cb7GxXr3DMk/Ue6rEu2j7JdPSr7JcX7tX6nWD//QJVLCau/hanOfYdiR+texUZvNXHQqxUbfssUf18fVzzbih6iPTRhTbWHFkg6N4Twu5a6/LOKB4uLWf6qQgirJL1I0qG2/7XNKOOtz+5On+0uxf37HamelGLCjIck3aK47XxV8fJfpb9fUby3/g+K38vfphgfVryX++epnn3WWD/fWKSz5C9XPCi0QvEM58vDukvplWJbmD7fxorbpyTJMUPjUWlejykezDxacbv+K8XbQh5L41a/x2HDZ6BaLyN8m2Kn8oRq/d0yzhFpOT/VGLj9JdsAAAAAgPHqqzNYAAAAAFf4w94AACAASURBVDCZ0cECAAAAgIbQwQIAAACAhtDBAgAAAICGdPVBw7bJqAF0WAihU88N22ANDQ2FefPmtS1bvHgiz7ker/0LZaU4ctN1M/a6Sp+5pO5na7+88ppvfj3uv3/7JXZ3e+u65SGE2b0OYrKZMmVKmDq1/TOE16xZ03Z4Z5SajKU4ctN1M/a66jaT63629ssrr/nm1+O0ae2X2N3trbuabjtNKIug7ZcqppWcKun/tTzost34dLCADqODFY2nfpo/f35YtGhRbj6dCbCtUhU53mfwjjZNv6j7s1D3s7VfXnnNN78ec7+93d3eum5xCGF+r4PotfG2naZPnx5mzZrVtmz58nbPX+2UoUJZKY7cdN2Mva7SZy6p+9naL6+85ptfj0ND7ZfY3e2tu5puO9W+RND2VEmfl3So4kPCjkwPmQOAnqJ+AtCPqJuAwTCRe7AOkHRTCOGW9FCvsxQffgkAvUb9BKAfUTcBA2AiHawdNPKJ3HekYSPYPtb2Itvtr78BgOaNWj9V66Zly5Z1NTgAA2vcbae1a9d2LTgAzeh4FsEQwskhhPlcdw2gn1Trptmzue8eQP+o1k9TppDwGZhsJrLX3ilpbuX9jmkYAPQa9ROAfkTdBAyAiaRp/5Wk3W0/VbFyeKOkNzUSFQBMzCStn+olMcolg50cCem6HWRueSS5bTWgmQ47bZLWTVLdzHiZhHSaHAnpuh1kbnl1sxluuPo902HtDlYIYY3t4yVdoJhq9LQQwjWNRQYANVE/AehH1E3AYJjQg4ZDCOdJOq+hWACgMdRPAPoRdROw4ePOSQAAAABoCB0sAAAAAGgIHSwAAAAAaAgdLAAAAABoyISSXADAhi6XqlrqRLrqQqrwOlnEi7NrPi150+ujFKObTu9eWh1dzEre3e2tvDz0v1yqaqkT6aoLqcLrZBEvzq75tORNr49SjMubTu9eWh1dzEre3e2tvLx+xxksAAAAAGgIHSwAAAAAaAgdLAAAAABoCB0sAAAAAGgIHSwAAAAAaAhZBAGgTzSd0K0TmQLrLi+X9a8YYc3Mft3MPljM+tfokjqTYbC0Prq9/aC/NZ3QrROZAusuL5f1rxhhzcx+3cw+WMz61+iSOpNhsLQ+ur39jBdnsAAAAACgIXSwAAAAAKAhdLAAAAAAoCF0sAAAAACgIXSwAAAAAKAhdLAAAAAAoCGkaQeAbupi5utS4u5yBvSaqbuLRZnCUnrxmqnH66ib5ryk3ldd91urqc7iaqbPxyTQxczXpcTd5QzoNVN3F4syhaX04jVTj9dRN815Sb2vuu63VlOdxdVMn980zmABAAAAQEPoYAEAAABAQ+hgAQAAAEBD6GABAAAAQEPoYAEAAABAQ+hgAQAAAEBDSNMOAN3U5SzcdRRTsTe+sMKy6qZOL6WLz8yyX7KLlz5y6Jco+yQMdECXs3DXUUzF3vjCCsuqmzq9lC4+M8suZhcvKn3koX6Jsk/CmFAHy/YSSaslPSFpTQhhfhNBAcBEUT8B6EfUTcCGr4kzWH8RQuiT/iIAjED9BKAfUTcBGzDuwQIAAACAhky0gxUk/cj2YtvHthvB9rG2F9leNMFlAcB4FOunat20bNmyHoQHYECNq+20du3aLocHYKImeongQSGEO20/RdKFtq8PIVxSHSGEcLKkkyXJdp/cwg1gABTrp2rdNH/+fOomAN0yrrbT9OnTqZ+ASWZCZ7BCCHemv/dK+rakA5oICgAmivoJQD+ibgI2fLXPYNneTNKUEMLq9P+LJZ3YWGQAUNOGWD/VyoxdzPnd/wfFO5INPPexJ0Hq8UkQIkaxIdZNUs3M2MWc332SE76gIxlKch97EqRDmQQhdtVELhHcVtK3HX/Ap0n6
```

Comment:
Can you move this into a function because the same code is also used above when creating the base model w/o adversarial learning? You can then invoke the same function in both places.

Reply:
Thanks this is done